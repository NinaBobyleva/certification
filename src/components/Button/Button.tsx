import * as S from "./button.styled.ts";

export function Button({
  title,
  onClick,
}: {
  title: string;
  onClick: () => void;
}) {
  return <S.Button onClick={onClick}>{title}</S.Button>;
}
