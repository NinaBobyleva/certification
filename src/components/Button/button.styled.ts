import styled from "styled-components";

export const Button = styled.button`
    width: 150px;
    height: 42px;
    border-radius: 30px;
    border: none;
    background-color: lightgrey;
    font-weight: 400;
    font-size: 14px;
    line-height: 1px;
    color: #000;
`;