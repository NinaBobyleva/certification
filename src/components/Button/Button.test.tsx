import renderer from "react-test-renderer";
import { Button } from "./Button";

describe("Компонент Button", () => {
  it("Успешный рендер Button", () => {
    const onClick = jest.fn();
    const component = renderer
      .create(<Button title="Следующий" onClick={onClick} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
