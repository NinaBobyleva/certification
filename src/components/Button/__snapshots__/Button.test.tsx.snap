// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент Button Успешный рендер Button 1`] = `
<button
  className="sc-blHHSb flYjWY"
  onClick={[MockFunction]}
>
  Следующий
</button>
`;
